acceptance/tests/resource/package/common_package_name_in_different_providers.rb

```diff
@@ -30,6 +32,17 @@
   end
 end
 
+# This is necessary so that the puppet-agent gem binary is in the path
+# during `puppet apply` runs. Otherwise, the gem provider is considered
+# unsuitable.
+def path
+  if @options[:type] == 'aio'
+    "#{agents[0]['privatebindir']}:${PATH}"
```

Comment:
This won't work (reliably) if the master has the agent role, and you're testing a windows agent.